libraries/botbuilder-repo-utils/src/updateVersions.ts

```diff
@@ -23,80 +23,65 @@ const getPackageVersion = (
     pkg: Package,
     newVersion: string,
     options: Record<'commitSha' | 'date' | 'deprecated' | 'preview', string | undefined>
-): string[] => {
-    let version: Array<string | undefined> = [newVersion];
+): string => {
+    const extra = [options.commitSha, options.date];
 
     if (pkg.deprecated) {
-        version.push(options.deprecated);
+        extra.unshift(options.deprecated);
     } else if (pkg.preview) {
-        version.push(options.preview);
+        extra.unshift(options.preview);
     }
 
-    return R.compact([...version, options.date, options.commitSha]);
+    return R.compact([newVersion, R.compact(extra).join('.')]).join('-');
```

Comment:
This refactor is to make versions more consistent with how they used to be: `4.11.0-dev.$DATE.$COMMIT`.